src/app/devices/components/DeviceHeaderForms/AddDeviceForm/types.ts

```diff
@@ -20,6 +20,6 @@ export type AddDeviceInterface = {
 export type AddDeviceValues = {
   domain: Domain["name"];
   hostname: Device["hostname"];
-  interfaces: AddDeviceInterface[];
+  interfaces: (Omit<AddDeviceInterface, "subnet"> & { subnet: string })[];
```

Comment:
I think we can modify the `AddDeviceInterface` directly and set the subnet to string, it's only used in this file.